pkg/sharings/sharings.go

```diff
@@ -597,6 +603,41 @@ func CreateSharing(instance *instance.Instance, sharing *Sharing) error {
 	return couchdb.CreateDoc(instance, sharing)
 }
 
+// RevokeSharing revokes the sharing and deletes all the OAuth client
+// associated with it.
+//
+// Revoking a sharing consists of setting the field `Revoked` to `true`.
+func RevokeSharing(ins *instance.Instance, sharing *Sharing) error {
+	sharing.Revoked = true
+
+	var err error
+	if sharing.Owner {
+		if sharing.Type == consts.MasterMasterSharing {
+			for _, recipient := range sharing.RecipientsStatus {
+				err = deleteOAuthClient(ins, recipient.HostClientID)
+			}
+		}
+	} else {
+		err = deleteOAuthClient(ins, sharing.Sharer.SharerStatus.HostClientID)
+	}
+
+	return err
+}
+
+func deleteOAuthClient(ins *instance.Instance, id string) error {
+	client, err := oauth.FindClient(ins, id)
+	if err != nil {
+		log.Errorf("[sharings] Could not find client %s: %v", id, err)
+		return err
+	}
+	crErr := client.Delete(ins)
```

Comment:
Because `client.Delete` returns a `ClientRegistrationError` structure not a simple error